usecases/cluster/transactions_write.go

```diff
@@ -128,19 +224,19 @@ func (c *TxManager) CommitWriteTransaction(ctx context.Context,
 ) error {
 	c.Lock()
 	if c.currentTransaction == nil || c.currentTransaction.ID != tx.ID {
+		expired := c.expired(tx.ID)
 		c.Unlock()
+		if expired {
+			return ErrExpiredTransaction
+		}
 		return ErrInvalidTransaction
 	}
 
 	c.Unlock()
 
 	// now that we know we are dealing with a valid transaction: no  matter the
 	// outcome, after this call, we should not have a local transaction anymore
-	defer func() {
-		c.Lock()
-		c.currentTransaction = nil
-		c.Unlock()
-	}()
+	defer c.clearTransaction()
```

Comment:
shouldn't that be inside `c.Lock()` / `c.Unlock()`? All other calls are

Reply:
Great catch, the locks are clearly missing. I'll add them.